homeassistant/components/command_line/sensor.py

```diff
@@ -137,8 +137,10 @@ async def async_update(self) -> None:
                     _LOGGER.warning("Unable to parse output as JSON: %s", value)
             else:
                 _LOGGER.warning("Empty reply found when expecting JSON data")
+            if self._value_template is None:
+                value = None
 
-        elif self._value_template is not None:
+        if self._value_template is not None:
```

Comment:
```suggestion
        elif self._value_template is not None:
```

If we use elif we know json attributes is None, so we just can assign the rendered template, and return.
The the case that this left is where we just assign the value. We do not need an if for that.

At L150

```python
            ...
        else:
            self._attr_native_value = value
 ```

becomes

```python
            ...
            return
        self._attr_native_value = value
 ```

Reply:
I think that would bring back the original issue where value_template would only be used if json_attributes isn't, but the two options should be usable together.

Reply:
Sorry, i'll have a better look at it

Reply:
You are right about the `if` vs `elsif`, so leave that, I updated the other suggestion.

Finally you get something like:

```python
    async def async_update(self) -> None:
        """Get the latest data and updates the state."""
        await self.hass.async_add_executor_job(self.data.update)
        value = self.data.value

        if self._json_attributes:
            self._attr_extra_state_attributes = {}
            if value:
                try:
                    json_dict = json.loads(value)
                    if isinstance(json_dict, Mapping):
                        self._attr_extra_state_attributes = {
                            k: json_dict[k]
                            for k in self._json_attributes
                            if k in json_dict
                        }
                    else:
                        _LOGGER.warning("JSON result was not a dictionary")
                except ValueError:
                    _LOGGER.warning("Unable to parse output as JSON: %s", value)
            else:
                _LOGGER.warning("Empty reply found when expecting JSON data")
            if self._value_template is None:
                self._attr_native_value = None
                return
            
        if self._value_template is not None:
            self._attr_native_value = (
                self._value_template.async_render_with_possible_json_value(
                    value,
                    None,
                )
            )
            return
        self._attr_native_value = value
```